airflow/contrib/operators/ssh_operator.py

```diff
@@ -71,11 +71,19 @@ def execute(self, context):
             if self.ssh_conn_id:
                 if self.ssh_hook and isinstance(self.ssh_hook, SSHHook):
                     self.log.info("ssh_conn_id is ignored when ssh_hook is provided.")
+                    
+                if self.remote_host is not None:
+                    self.ssh_hook.remote_host = self.remote_host
+
                 else:
                     self.log.info("ssh_hook is not provided or invalid. " +
                                   "Trying ssh_conn_id to create SSHHook.")
-                    self.ssh_hook = SSHHook(ssh_conn_id=self.ssh_conn_id,
-                                            timeout=self.timeout)
+                    if self.remote_host is not None:
```

Comment:
We don't need this condition - passing None is the same as not passing it at all in this case.

Reply:
good point.